### PR TITLE
RHCLOUD-23554 Add text in UI to inform user that an email notif is being forced

### PR DESCRIPTION
**Pull request by aferd**
Screen Shot 2023-04-06 at 1 23 35 PM
this is what it would look like when there is a forced email. there is currently NO apps sending forced emails so you will not see this in the UI as of now.

### CODE REVIEW

File: src/components/Notifications/Recipient.tsx [hunk 1]
```diff
@@ -27,6 +27,9 @@ const greyColorName = style({
 const CommaSeparator = () => <span>, </span>;
 
 export const Recipient: React.FunctionComponent<RecipientProps> = (props) => {
+
+    const [ ignorePref ] = React.useState<EmailSystemProperties>();
```
Comment (josejulio) [inline src/components/Notifications/Recipient.tsx]: I think you still need to load this value into the `Action`.

Here is how to do it:

1. Update [NotificationAdapter](https://github.com/josejulio/notifications-frontend/blob/f711c5d03c6f66372489b3b44c9355cbc1b2fc5b/src/types/adapters/NotificationAdapter.ts#L35) to use the ignore preferences variable (`integration.ignorePreferences`) and pass it to `NotificationUserRecipient`
2. Update [NotificationUserRecipient](https://github.com/josejulio/notifications-frontend/blob/ffc76263cfc72f160bc31f693c3ffbeaa5d395b2/src/types/Recipient.ts#L56) to receive a third boolean for ignorePreferences, and assign it accordingly. You will need to update other places where it's called.
3.  The setting is per recipient. i.e. the recipient "all users" might not be forced but "send to admins" might be forced so I suggest to update:
`Users: {join(users.map(u => u.displayName), CommaSeparator)}` to something like:
```
Users: {join(users.map(u => <>
  {u.displayName}
  {u.ignoreUserPreferences && <span className={ greyColorName }> { props.action.integration.name }
                        <Tooltip content="This integration has been disabled. This action wil not fire until it is enabled." position="bottom">
                            <BanIcon className={ disabledLabelClassName } />
                        </Tooltip>
                    </span>)
</>), CommaSeparator)}
```
The lock icon should show only for the users that are forced.

Alternatively you could display it on the end by checking if there is at least one forced recipient:
```
const ignoreUserPreferences = users.some(u => u.ignoreUserPreferences)
```

4. Theorically the groups could also be forced, but there is no way to set it right now.

**Review by josejulio — CHANGES_REQUESTED** [review]

**Comment by aferd** [comment]
@josejulio when you get time can you re-review? thanks!

File: src/types/Recipient.ts [hunk 2]
```diff
@@ -72,6 +74,7 @@ export class NotificationUserRecipient extends BaseNotificationRecipient {
         );
 
         this.sendToAdmin = sendToAdmin;
+        this.ignorePreferences = ignorePreferences;
     }
 
     public equals(recipient: Recipient) {
```
Comment (josejulio) [inline src/types/Recipient.ts:80]: Could you update this equal to include the `ignorePreferences`? 

Change `return recipient.sendToAdmin === this.sendToAdmin;` to `return recipient.sendToAdmin === this.sendToAdmin && recipient.ignorePreferences === this.ignorePreferences;`

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**